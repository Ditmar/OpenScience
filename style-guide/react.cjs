const { resolve } = require('node:path');
const project = resolve(process.cwd(), 'tsconfig.json');

module.exports = {
  env: {
    browser: true,
    es2021: true,
    es6: true,
    jest: true,
  },
  extends: [
    require.resolve('./rules/base'),
    require.resolve('./configurations/_base'),
    'plugin:react-hooks/recommended',
    'airbnb',
    'plugin:storybook/recommended',
    require.resolve('./configurations/typescript'),
    require.resolve('./configurations/vitest'),
  ],
  parserOptions: {
    project,
    babelOptions: {
      presets: ['@babel/preset-react'],
    },
  },
  plugins: ['react-refresh'],
  settings: {
    react: {
      version: 'detect',
    },
  },
  globals: {
    JSX: true,
  },
  rules: {
    'react/react-in-jsx-scope': 'off',
    'react/button-has-type': 'off',
    'react-refresh/only-export-components': [
      'warn',
      { allowExportNames: ['meta', 'links', 'headers', 'loader', 'action'] },
    ],
  },
};
