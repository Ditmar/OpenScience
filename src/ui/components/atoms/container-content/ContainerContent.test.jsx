import { render } from '@testing-library/react';
import Container from './ContainerContent.tsx';

test('renders container with initial state', async () => {
  const { container } = render(Container);
  expect(container).toMatchSnapshot();
});
