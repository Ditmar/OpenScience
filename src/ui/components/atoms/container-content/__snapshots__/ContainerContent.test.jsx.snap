// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders container with initial state 1`] = `<div />`;
